### ISSUE TITLE
Levitating entity when using RelativeObjectPosition

### ISSUE THREAD

**Issue by d-hir** [esmini/esmini#509]
The attached OpenSCENARIO and OpenDRIVE files ([levitating_car.zip](https://github.com/esmini/esmini/files/13728963/levitating_car.zip)) are almost identical to esmini's `straight_500m.xosc`/`.xodr` examples. The only differences are:
 - I removed the dependency on other files (removed the vehicle catalog and removed the model3d reference)
 - Entity `Target` uses a `RelativeObjectPosition` relative to `Ego` for positioning
 - I added one more entity `Target2` which is identical to `Target` and is positioned relative to `Target` via `RelativeObjectPosition`
 - The OpenDRIVE file now has a non-zero elevation (linearly rising in +x direction, which is also the direction of the cars)

Identical to esmini's `straight_500m.xosc` scenario, this scenario orders the cars to simply drive on a straight road.

**What I would expect** when running this scenario without any external input:
 - All three cars are driving normally on the road, one after another, all separated by the 25 m that I set in the xosc file

### Version: 2.34.1

**What actually happens**:
 - The `Ego` and the `Target` behave as expected and drive on the road
 - `Target2` (which is the front-most car) levitates above the road
 
![scene](https://github.com/esmini/esmini/assets/130217846/cc43bd32-0cdc-459f-93e5-8c05bd8e6d2e)
^ Total scene: Right-most car is `Ego`, middle one is `Target`, levitating one is `Target2`

![closeup](https://github.com/esmini/esmini/assets/130217846/c34806e3-55b0-487c-9002-658715a5d3db)
^ Close-up shot of the levitating `Target2` and normal `Target`

![closeside](https://github.com/esmini/esmini/assets/130217846/b53eef0f-ef9d-4a16-b97d-881e5c1cfb67)
^ Side view of `Target2`, also note the wrong pitch of the car (the road is rising to the left, so the car is pointing even more up!)

![sideview](https://github.com/esmini/esmini/assets/130217846/6ef2d108-0754-4121-a311-5713127cfe35)
^ Total scene from the side (from left to right: `Target2`, `Target`, `Ego`)

![topdown](https://github.com/esmini/esmini/assets/130217846/3743a878-914a-4ab6-aa2d-7accf541c98a)
^ Top-down view (from left to right: `Target2`, `Target`, `Ego`)

Logfile: [esmini.log](https://github.com/esmini/esmini/files/13729248/esmini.log)
OS: Win10

Changing the position to a `RelativeLanePosition` fixes this.
Removing the elevation from the OpenDRIVE file does as well.

### Version: 2.35.0

I also just saw that there is already a version 2.35.0, so I tried it with that one. But here it is even worse:

**What actually happens**:
 - The `Ego` behaves as expected and drives on the road
 - `Target` levitates and also points slightly upwards, away from the road
 - `Target2` levitates even higher and is even more pitched upwards

![scene_2_35_0](https://github.com/esmini/esmini/assets/130217846/f67fe6a3-4384-4778-8054-a8e3e40f672a)
^ Total scene in 2.35.0

![sideview_2_35_0](https://github.com/esmini/esmini/assets/130217846/f176dd68-8da4-4696-8fd6-366854261d65)
^ Total side view

![close_sideview_2_35_0](https://github.com/esmini/esmini/assets/130217846/99433a0a-9498-4863-b5a4-9d9e34bd53a5)
^ Close-up side view

Logfile: [esmini_2_35_0.log](https://github.com/esmini/esmini/files/13729383/esmini_2_35_0.log)

**Comment by eknabevcc** [comment]
Thanks for reporting this finding. You're totally right. In previous esmini versions the relative positioning had some limitations and issues. 

Improvements has been implemented for release [v2.36.0](https://github.com/esmini/esmini/releases/tag/v2.36.0). 

The relative position should now consider the complete orientation of the referenced entity, not only heading.

My initial test with your provided scenario results in the image below, actually I increased the slope from 0.03 to 0.5 to clarify.

![image](https://github.com/esmini/esmini/assets/29146839/edf1ee01-5852-46db-8dde-3cb9a8d5bcd5)
